OMCompiler/CMakeLists.txt

```diff
@@ -1,47 +1,23 @@
-# Jens Frenkel, Jens.Frenkel@tu-dresden.de, 2010-03-30
-# CMakefile for compilation of OMC
+cmake_minimum_required(VERSION 3.14)
```

Comment:
It's a big problem for nightly builds and Jenkins, and especially source packages

Reply:
@AnHeuermann About CMake 3.14, I understand absolutely what you mean. It will be a hassle to get recent CMake versions on long term releases of some Linux distros. The decision I had to make was whether to use old cmake and repeat the same "quirks" of CMake that everyone hated to begin with OR just go for the new and improved cmake and have something that can be somehow future proof.  

This definitely needs some READMEs and proper documentation. I will do that soon. It is not over yet at all. The MetaModelica compilation still needs some work. There are issues with some 3rdParty libs and many other problems. I t was just that I have other things to fix right now. It is also on a relatively stable stage now which means you can for example start adding your won libraries to this. I will try to add some documentation about that. However, the good thing about these recent version of cmake is that it should be very straight forward to add new libraries. Take a look at some of the libraries in simulation runtime. It will make sense quickly. I will add this on the documentation as well.

There is still a lot left to do.

Reply:
@sjoelund You do not have to use it on Jenkins if it is a big problem.

Reply:
@AnHeuermann 
> There are some files in OMCompiler/Compiler/.cmake/unused/. Do they have any meaning?
That was an attempt to add MetaModelica as a kind of "native" language to CMake. It did not work out as good as expected for many reasons. I had it in my branch and was not removed. I will remove it.

Reply:
@AnHeuermann 
> E.g. if I would generate a new MetaModelica file crazyNewStuff.mo where I would need to add it so that it get's compiled?

Hopefully you only have to add it one place when this is over. That would be at `OMCompiler/Compiler/.cmake/meta_modelica_source_list.cmake`

Right now you need to the package name to `OMCompiler/Compiler/.cmake/package_list.txt` but that won't be necessary later.